pkg/store/copr/coprocessor.go

```diff
@@ -850,9 +857,16 @@ func (worker *copIteratorWorker) run(ctx context.Context) {
 }
 
 // open starts workers and sender goroutines.
-func (it *copIterator) open(ctx context.Context, enabledRateLimitAction, enableCollectExecutionInfo bool) {
+func (it *copIterator) open(ctx context.Context, tryCopLiteWorker *uint32) {
+	if len(it.tasks) == 1 && tryCopLiteWorker != nil && atomic.LoadUint32(tryCopLiteWorker) == 0 {
+		atomic.StoreUint32(tryCopLiteWorker, 1)
```

Comment:
This is not atomic operation, the `CAS` should used instead.

But why it's designed to be read-write concurrently? Should the `open` be a single-threaded operation?

Reply:
In some cases (such as index lookup reader), there are maybe some `TableReader` execute `open` concurrently.

Reply:
The workers under the same session is individual, I think the lite worker can be used in different `TableReader`s under one index lookup executor. However, the `DistSQLContext` is shared in session level.

By using `if atomic.CompareAndSwapUint32(tryCopLiteWorker, 0, 1) {...}`, only one cop iterator can benefit from the lite optimization.

Reply:
@crazycs520 
Is it possible to make the lite worker to be used for all the table workers?